c10/core/TensorImpl.cpp

```diff
@@ -292,33 +300,44 @@ bool _compute_channels_last_contiguous_2d(
   }
 }
 
-bool TensorImpl::compute_channels_last_contiguous_2d(identity<bool>) const {
+SymBool TensorImpl::compute_channels_last_contiguous_2d(
+    identity<SymBool>) const {
   if (is_sparse()) {
     return false;
   }
-  return _compute_channels_last_contiguous_2d<int64_t>(
-      sizes_and_strides_.sizes_arrayref(),
-      sizes_and_strides_.strides_arrayref());
+  SymIntArrayRef sizes = extra_meta_->sizes_;
+  SymIntArrayRef strides = extra_meta_->strides_;
+  switch (sizes.size()) {
+    case 4: {
+      SymInt expected = 1;
+      SymBool r = true;
+      for (auto& d : {1, 3, 2, 0}) {
+        const auto& size_d = sizes[d];
+        r = r & (size_d.sym_eq(1) | strides[d].sym_eq(expected));
+        expected *= size_d;
+      }
+      return r;
+    }
+    // NOLINTNEXTLINE(bugprone-branch-clone)
+    case 3:
+      // TODO dim == 3 case will be enabled once it is fully tested
+      return false;
```

Comment:
hmm what does this mean? (Why do we always want to report this case as non-contiguous until it's fully tested?)

Reply:
This is from the old code. What I believe happened was the original memory format workstream was going to change the definition of contiguity to also handle the dim 3 case, but people never got around to doing it.